lib/src/messages/logical_replication_messages.dart

```diff
@@ -370,32 +390,43 @@ class TupleData {
   /// TupleData does not consume the entire bytes
   ///
   /// It'll read until the types are generated.
-  factory TupleData._parse(PgByteDataReader reader) {
+  factory TupleData._parse(PgByteDataReader reader, int relationId) {
     final columnCount = reader.readUint16();
     final columns = <TupleDataColumn>[];
     for (var i = 0; i < columnCount; i++) {
       // reading order matters
       final typeId = reader.readUint8();
+      final typeOid = RelationMessage._getTypeOid(relationId, i);
       final tupleDataType = TupleDataType.fromByte(typeId);
       late final int length;
       late final String data;
+      late final Object? value;
       switch (tupleDataType) {
         case TupleDataType.text:
-        case TupleDataType.binary:
           length = reader.readUint32();
           data = reader.encoding.decode(reader.read(length));
+          value = null;
+          break;
+        case TupleDataType.binary:
+          length = reader.readUint32();
+          value = TypeRegistry().decodeBytes(reader.read(length),
```

Comment:
Let's not create a new `TypeRegistry` instance, rather get a reference to it from the `Settings` object (similarly to the way we get a reference to the encoding). This change itself should be in a separate PR, and I may give a hand in it, if I will have some time today evening.